spec/fixed_fields/bad_008_spec.rb

```diff
@@ -5,19 +5,87 @@
+      ]
+    end
+    let(:leader) { '01104naa a2200289 i 4500' }
+    let(:record) { MARC::Record.new_from_hash('fields' => fields, 'leader' => leader) }
     it 'knows that a record with bad book 008 is invalid' do
-      expect(MarcCleanup.bad_008?(record_with_bad_008)).to eq true
+      expect(MarcCleanup.bad_008?(record)).to eq true
     end
   end
   describe 'bad computer format 008' do
-    let(:marcfile) {"#{Dir.getwd}/spec/fixtures/fixed_fields/marc_with_bad_comp_008.xml"}
-    let(:reader) { MARC::XMLReader.new(marcfile, parser: 'magic') }
-    let(:record_with_bad_008) { reader.first }
+    let(:fields) do
+      [
+        { '008' => '230519s1996    njua   u      000 0 eng d' }
+      ]
+    end
+    let(:leader) { '01104nma a2200289 i 4500' }
+    let(:record) { MARC::Record.new_from_hash('fields' => fields, 'leader' => leader) }
     it 'knows that a record with bad computer 008 is invalid' do
-      expect(MarcCleanup.bad_008?(record_with_bad_008)).to eq true
+      expect(MarcCleanup.bad_008?(record)).to eq true
+    end
+  end
+  describe 'bad map format 008' do
+    let(:fields) do
+      [
+        { '008' => '230519s1996    njua   u      000 0 eng d' }
+      ]
+    end
+    let(:leader) { '01104nea a2200289 i 4500' }
+    let(:record) { MARC::Record.new_from_hash('fields' => fields, 'leader' => leader) }
+    it 'knows that a record with bad map 008 is invalid' do
+      expect(MarcCleanup.bad_008?(record)).to eq true
+    end
+  end
+  describe 'bad music format 008' do
+    let(:fields) do
+      [
+        { '008' => '230519s1996    njua   u      000 0 eng d' }
+      ]
+    end
+    let(:leader) { '01104nca a2200289 i 4500' }
+    let(:record) { MARC::Record.new_from_hash('fields' => fields, 'leader' => leader) }
+    it 'knows that a record with bad music 008 is invalid' do
+      expect(MarcCleanup.bad_008?(record)).to eq true
+    end
+  end
+  describe 'bad continuing resource format 008' do
+    let(:fields) do
+      [
+        { '008' => '230519s1996    njua   u      000 0 eng d' }
+      ]
+    end
+    let(:leader) { '01104nab a2200289 i 4500' }
+    let(:record) { MARC::Record.new_from_hash('fields' => fields, 'leader' => leader) }
+    it 'knows that a record with bad continuing resource 008 is invalid' do
+      expect(MarcCleanup.bad_008?(record)).to eq true
+    end
+  end
+  describe 'bad visual format 008' do
+    let(:fields) do
+      [
+        { '008' => '230519s1996    njua   u      000 0 eng d' }
+      ]
+    end
+    let(:leader) { '01104nga a2200289 i 4500' }
+    let(:record) { MARC::Record.new_from_hash('fields' => fields, 'leader' => leader) }
+    it 'knows that a record with bad visual 008 is invalid' do
+      expect(MarcCleanup.bad_008?(record)).to eq true
+    end
+  end
+  describe 'bad mix_mat format 008' do
+    let(:fields) do
+      [
+        { '008' => '230519s1996    njua   u      000 0 eng d' }
+      ]
+    end
+    let(:leader) { '01104nga a2200289 i 4500' }
```

Comment:
This leader and the 008 are the same as the ones for the visual format. Positions 6-7 need to be one of:
pa
pb
pc
pd
pi
pm
ps